### PR TITLE
Fix the creeps setting being ignored

### PR DESCRIPTION
**Pull request by abcdefg30**
Fixes @obrakmann's find: https://github.com/OpenRA/OpenRA/pull/8744#discussion_r36602685.

### CODE REVIEW

File: mods/cnc/maps/gdi05b/map.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@ Options:
 	StartingCash: 4000
 	ConfigurableStartingUnits: False
 	ShortGame: False
+	Creeps: False
```
Comment (Mailaender) [inline mods/cnc/maps/gdi05b/map.yaml]: > System.ArgumentException: An item with the same key has already been added.

File: OpenRA.Mods.RA/Traits/SpawnActorOnDeath.cs [hunk 2]
```diff
@@ -42,6 +42,9 @@ public class SpawnActorOnDeathInfo : ITraitInfo
 		[Desc("Skips the husk actor's make animations if true.")]
 		public readonly bool SkipMakeAnimations = true;
 
+		[Desc("Should no husk been spawned when the 'Creeps' setting is false?")]
```
Comment (pchote) [inline OpenRA.Mods.RA/Traits/SpawnActorOnDeath.cs]: Can you please fix all these references to 'husks'?  They shouldn't have survived the initial renaming to `SpawnActorsOnDeath`.

File: OpenRA.Mods.RA/Traits/SpawnActorOnDeath.cs [hunk 3]
```diff
@@ -42,6 +42,9 @@ public class SpawnActorOnDeathInfo : ITraitInfo
...
+		public readonly bool AffectedByCreepsSetting = false;
```
Comment (pchote) [inline OpenRA.Mods.RA/Traits/SpawnActorOnDeath.cs]: `RequiresLobbyCreeps` might be a bit more concise.

**Comment by abcdefg30** [comment]
Updated.

**Comment by Mailaender** [comment]
Thanks. :+1:

File: OpenRA.Mods.RA/Traits/SpawnActorOnDeath.cs [hunk 4]
```diff
@@ -25,23 +25,26 @@ public class SpawnActorOnDeathInfo : ITraitInfo
 		[Desc("Actor to spawn on death.")]
 		public readonly string HuskActor = null;
```
Comment (pchote) [inline OpenRA.Mods.RA/Traits/SpawnActorOnDeath.cs]: Sorry to keep on pushing, but can you fix this one too?

**Comment by abcdefg30** [comment]
Updated and renamed `HuskActor` to `Actor`.

File: OpenRA.Mods.RA/Traits/SpawnActorOnDeath.cs [hunk 5]
```diff
@@ -23,25 +23,28 @@ public class SpawnActorOnDeathInfo : ITraitInfo
...
-			"Leave empty to spawn a husk actor ignoring the DeathTypes.")]
+		[Desc("DeathType that triggers the actor spawn." +
+			"Leave empty to spawn an actor ignoring the DeathTypes.")]
 		public readonly string DeathType = null;
 
-		[Desc("Skips the husk actor's make animations if true.")]
+		[Desc("Skips the spawned actor's make animations if true.")]
 		public readonly bool SkipMakeAnimations = true;
 
+		[Desc("Should no actor been spawned when the 'Creeps' setting is false?")]
```
Comment (penev92) [inline OpenRA.Mods.RA/Traits/SpawnActorOnDeath.cs]: This was a bit hard to stomach at first because of all the negativity.
Also "be" instead of "been".

**Comment by penev92** [comment]
:+1: with the above comment.

**Comment by abcdefg30** [comment]
Updated.

**Comment by penev92** [comment]
Thanks.

**Comment by penev92** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/c53b52c1fa5aaa3993a78f9fc2978bae638d8df6...6094c3120b9dc7e1664c7aeb0aebcecf2bcc01be

**Pull request merged**